### PR TITLE
Docs: Remove links to external C functions and macros in os.rst

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
:books: Documentation preview :books:: https://cpython-previews--108138.org.readthedocs.build/

### CODE REVIEW

**Comment by erlend-aasland** [comment]
The amount of Sphinx warnings in os.rst is huge. I figured it was better to fix them in chunks, starting with the external C functions and macros.

**Comment by erlend-aasland** [comment]
This is a really trivial change; I think it would be ok to merge it as soon as the CI is gree. But, I'll await Adam's review, now that I requested it.

File: Doc/library/os.rst [hunk 1]
```diff
@@ -3147,14 +3147,16 @@ features:
...
       ``BY_HANDLE_FILE_INFORMATION`` structure returned by
-      :c:func:`GetFileInformationByHandle`. See the ``FILE_ATTRIBUTE_*``
+      :c:func:`!GetFileInformationByHandle`.
+      See the :const:`!FILE_ATTRIBUTE_* <stat.FILE_ATTRIBUTE_ARCHIVE>`
       constants in the :mod:`stat` module.
 
    .. attribute:: st_reparse_tag
 
-      When :attr:`st_file_attributes` has the ``FILE_ATTRIBUTE_REPARSE_POINT``
+      When :attr:`st_file_attributes` has the :const:`FILE_ATTRIBUTE_REPARSE_POINT`
```
Comment (AA-Turner) [inline Doc/library/os.rst]:
```suggestion
      When :attr:`st_file_attributes` has the :const:`~stat.FILE_ATTRIBUTE_REPARSE_POINT`
```

**Review by AA-Turner — APPROVED** [review]
One note:

A

File: Doc/library/os.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/library/os.rst]: Ah, thanks!

**Pull request merged**